src/azure-cli/azure/cli/command_modules/acs/decorator.py

```diff
@@ -0,0 +1,570 @@
+    if isinstance(li, list):
+        try:
+            return li[idx]
+        except IndexError:
+            return default
+    return None
+
+
+# pylint: disable=too-many-instance-attributes,too-few-public-methods
+class AKSCreateModels:
+    # Used to store models (i.e. the corresponding class of a certain api version specified by `resource_type`)
+    # which would be used during the creation process.
+    def __init__(
+        self,
+        cmd: AzCommandsLoader,
+        resource_type: ResourceType = ResourceType.MGMT_CONTAINERSERVICE,
+    ):
+        self.cmd = cmd
+        self.resource_type = resource_type
+        self.ManagedClusterWindowsProfile = self.cmd.get_models(
+            "ManagedClusterWindowsProfile",
+            resource_type=self.resource_type,
+            operation_group="managed_clusters",
+        )
+        self.ManagedClusterSKU = self.cmd.get_models(
+            "ManagedClusterSKU",
+            resource_type=self.resource_type,
+            operation_group="managed_clusters",
+        )
+        self.ContainerServiceNetworkProfile = self.cmd.get_models(
+            "ContainerServiceNetworkProfile",
+            resource_type=self.resource_type,
+            operation_group="managed_clusters",
+        )
+        self.ContainerServiceLinuxProfile = self.cmd.get_models(
+            "ContainerServiceLinuxProfile",
+            resource_type=self.resource_type,
+            operation_group="managed_clusters",
+        )
+        self.ManagedClusterServicePrincipalProfile = self.cmd.get_models(
+            "ManagedClusterServicePrincipalProfile",
+            resource_type=self.resource_type,
+            operation_group="managed_clusters",
+        )
+        self.ContainerServiceSshConfiguration = self.cmd.get_models(
+            "ContainerServiceSshConfiguration",
+            resource_type=self.resource_type,
+            operation_group="managed_clusters",
+        )
+        self.ContainerServiceSshPublicKey = self.cmd.get_models(
+            "ContainerServiceSshPublicKey",
+            resource_type=self.resource_type,
+            operation_group="managed_clusters",
+        )
+        self.ManagedClusterAADProfile = self.cmd.get_models(
+            "ManagedClusterAADProfile",
+            resource_type=self.resource_type,
+            operation_group="managed_clusters",
+        )
+        self.ManagedClusterAgentPoolProfile = self.cmd.get_models(
+            "ManagedClusterAgentPoolProfile",
+            resource_type=self.resource_type,
+            operation_group="managed_clusters",
+        )
+        self.ManagedClusterIdentity = self.cmd.get_models(
+            "ManagedClusterIdentity",
+            resource_type=self.resource_type,
+            operation_group="managed_clusters",
+        )
+        # pylint: disable=line-too-long
+        self.ComponentsQit0EtSchemasManagedclusterpropertiesPropertiesIdentityprofileAdditionalproperties = self.cmd.get_models(
```

Comment:
switch to the new sdk?

Reply:
also Components1Umhcm8SchemasManagedclusteridentityPropertiesUserassignedidentitiesAdditionalproperties

Reply:
Sure, done in commit [7f75d9b](https://github.com/Azure/azure-cli/pull/19259/commits/7f75d9be8f02597676dae06627b1c90cbe801329).